Start building a Cycloid Drawer

peterjmartinson opened the pull request:
DON'T APPROVE THIS YET!!
Why doesn't it work?

1. Get a circle to apparently roll back and forth
2. Get it to draw a cycloid (but not)

I can't tell what the problem is!  Trying to get the cycloid to be drawn from an array of positions, but JS is freaking out about trying to add points.

pmartinson-dkp committed:
Start building a Cycloid Drawer

1. Get a circle to apparently roll back and forth
2. Get it to draw a cycloid (but not)

I can't tell what the problem is!  Trying to get the cycloid to be drawn
from an array of positions, but JS is freaking out about trying to add
ponts.

belcurv commented on src/features/draw-rolling-circle/Drawing.js:
```diff
@@ -0,0 +1,108 @@
+    ctx.save()
+
+    // fill in background, just for reference
+    ctx.fillStyle = Color.BLACK
```
You'll need to add `BLACK` to `/config.colors.js`

peterjmartinson replied:
Funny - it works!  I'll do this though.

belcurv commented on src/features/draw-rolling-circle/Drawing.js:
```diff
@@ -0,0 +1,108 @@
+    for (let i = 1; i < cycloid.i; i++) {
+      ctx.lineTo(cycloid.x[i], cycloid.y[i])
+    }
+    ctx.strokeStyle = Color.Yellow
```
```suggestion
    ctx.strokeStyle = Color.YELLOW
```

peterjmartinson replied:
Same - it works!

peterjmartinson replied:
Looks like Canvas ignores this, and just retains the last good strokeStyle setting, which is RED in this case.

belcurv commented on src/features/draw-rolling-circle/Drawing.js:
```diff
@@ -0,0 +1,108 @@
+      rollingCircle.centerX += rollingCircle.speed * radius * rollingCircle.direction
+      cycloid.i++
+      cycloid.x.push(pointX)
+      cycloid.Y.push(pointY)
```
Lowercase `y`

```suggestion
      cycloid.y.push(pointY)
```

peterjmartinson replied:
that's it.  Good eye.

belcurv commented on src/features/draw-rolling-circle/Drawing.js:
```diff
@@ -0,0 +1,108 @@
+const HEIGHT = 100
+const WIDTH = 500
+const radius = 30
+let pointX, pointY
```
Based on the code so far, I don't think we need to store these outside of the draw factory's / draw function's scope.

Down inside `draw`, you can 

```js
    // draw the line
    const pointX = rollingCircle.centerX + radius * Math.cos(rollingCircle.theta)
    const pointY = rollingCircle.centerY + radius * Math.sin(rollingCircle.theta)
```

peterjmartinson replied:
Cleaned up.

pmartinson-dkp committed:
Add BLACK to the colors

pmartinson-dkp committed:
Get the cycloid to draw

1. Fix the CaPiTaLiZaTiOn
2. Prevent the cycloid plot array from growing without bound
3. Resize everything to draw only one complete cycloid
